tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,193 @@
+"""
+Test osd node balancing by adding nodes and osds and checking their distribution
+"""
+import logging
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers
+from ocs_ci.ocs import constants, machine, ocp
+from ocs_ci.ocs.node import add_new_node_and_label_it, get_osd_running_nodes, get_nodes
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.utility.utils import (
+    ceph_health_check,
+    TimeoutSampler,
+    ocsci_log_path,
+)
+
+REPORT = "Node_and_Osd_distribution_report.txt"
+FINAL_REPORT = "Final Report"
+INITIAL_SETUP = "Initial Setup"
+MAX_NODE_COUNT = 9
+MAX_OSDS_PER_NODE = 3
+START_NODE_NUM = 3
+REPLICA_COUNT = 3
+OSD_LIMIT_AT_START = MAX_OSDS_PER_NODE * START_NODE_NUM
+NUM_NODE_ADDS = MAX_NODE_COUNT // REPLICA_COUNT
+
+
+def add_some_worker_nodes():
+    """
+    Add three worker nodes to the storage cluster.  Update the report file.
+    """
+    wnodecnt = len(get_nodes(constants.WORKER_MACHINE))
+    if wnodecnt == MAX_NODE_COUNT:
+        return
+    osd_running_nodes = get_osd_running_nodes()
+    logging.info(f"OSDs are running on nodes {osd_running_nodes}")
+    # Get the machine name using the node name
+    machine_names = [
+        machine.get_machine_from_node_name(osd_node) for osd_node in osd_running_nodes
+    ]
+    logging.info(f"{osd_running_nodes} associated " f"machine are {machine_names}")
+    # Get the machineset name using machine name
+    machineset_names = [
+        machine.get_machineset_from_machine_name(machine_name)
+        for machine_name in machine_names
+    ]
+    logging.info(f"{osd_running_nodes} associated machineset ")
+    node_obj = ocp.OCP(kind="node")
+    # This next section makes a new node on which OSD pods can run
+    for mnames in range(0, NUM_NODE_ADDS):
+        # This next section makes a new node on which OSD pods can run
+        new_node = add_new_node_and_label_it(
+            machineset_names[mnames], mark_for_ocs_label=False
+        )
+        node_obj.add_label(
+            resource_name=new_node[0], label=constants.OPERATOR_NODE_LABEL
+        )
+    collect_stats("Three nodes have been added")
```

Comment:
should it be variable (`NUM_NODE_ADDS`) ?

Reply:
Yes.  I will change this.

Reply:
Yes.  I'll fix this.